Source/Diagnostics/Diagnostics.H

```diff
@@ -18,99 +18,84 @@ class Diagnostics
 {
 public:
     Diagnostics (int i, std::string name);
-    ~Diagnostics ();
+    /** Virtual Destructor to handle clean destruction of derived classes */
+    virtual ~Diagnostics () ;
     /** Pack (stack) all fields in the cell-centered output MultiFab m_mf_output.
-     * Fields are computed (e.g., cell-centered) on-the-fly using a functor.*/
+     * Fields are computed (e.g., cell-centered or back-transformed)
+       on-the-fly using a functor.*/
     void ComputeAndPack ();
     /** Flush m_mf_output and particles to file. */
-    void Flush ();
-    /** Flush raw data */
-    void FlushRaw ();
+    virtual void Flush (){};
```

Comment:
```suggestion
    /** \brief Flush particle and field buffers to file using the FlushFormat member variable.
     *
     * This function should belong to class Diagnostics and not be virtual, as it flushes
     * the particle buffers (name not found yet) and the field buffers (m_mf_output), both
     * of which are members of Diagnostics. Yet, the implementation is left to derived classes
     * for now because:
     * - The functions underlying FlushFormat::WriteToFile expect a geometry object, which is
     *   WarpX::geom for FullDiagnostics but should be constructed for BTDiagnostics;
     * - The functions underlying FlushFormat::WriteToFile do not support writing a buffer to file
     *   multiple times yet.
     * When these are fixed, the implementation of Flush should be in Diagnostics.cpp
     */
    virtual void Flush (){};
```